src/main/java/jdbchomework/entity/AbstractEntity.java

```diff
@@ -22,11 +22,20 @@ public AbstractEntity() {
     }
 
     public AbstractEntity(long id, String name) {
+        if (name == null || name.equals("")) {
+            name = "DEFAULT";
+        }
+        if (id == 0) {
+            throw new IllegalArgumentException("Id must be > 0");
+        }
         this.id = id;
         this.name = name;
     }
 
     public AbstractEntity(String name) {
+        if (name == null || name.equals("")) {
+            name = "DEFAULT";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "name". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1226)